dvol_python/test_dvol.py

```diff
@@ -55,43 +80,52 @@ class VoluminousTests(TestCase):
     def setUp(self):
         self.tmpdir = FilePath(self.mktemp())
         self.tmpdir.makedirs()
-        self.patch(Voluminous, "lockFactory", NullLock)
 
     def test_create_volume(self):
-        # TODO test volume names with '/' in them - they should not end up
-        # making nested heirarchy
         dvol = VoluminousOptions()
-        dvol.parseOptions(["-p", self.tmpdir.path, "init", "foo"])
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
         self.assertTrue(self.tmpdir.child("foo").exists())
         self.assertTrue(self.tmpdir.child("foo").child("branches")
                 .child("master").exists())
-        self.assertEqual(dvol.voluminous.getOutput(),
-                ["Created volume foo", "Created branch foo/master"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Created volume foo\nCreated branch foo/master")
 
     def test_create_volume_already_exists(self):
         dvol = VoluminousOptions()
-        dvol.parseOptions(["-p", self.tmpdir.path, "init", "foo"])
-        self.assertRaises(VolumeAlreadyExists,
-                dvol.parseOptions, ["-p", self.tmpdir.path, "init", "foo"])
-        self.assertEqual(dvol.voluminous.getOutput(),
-                ["Error: volume foo already exists"])
+        # Create the repository twice, second time should have the error
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Error: volume foo already exists")
+
+    def test_create_volume_with_path_seperator(self):
+        dvol = VoluminousOptions()
+        dvol.parseOptions(ARGS + ["-p", self.tmpdir.path, "init", "foo/bar"])
+        self.assertEqual(dvol.voluminous.getOutput()[-1],
+                "Error: foo/bar is not a valid name")
```

Comment:
Rather than test for just this specific case, you could catch many more by doing something like:

```python

def invalid_volume_names():
    # XXX: Maybe extract a `valid_name` helper?
    return text().filter(lambda name: set(name) != set(letters))

@given(invalid_name=invalid_names())
def test_create_volume_with_invalid_name(self, invalid_name):
    # ... as above, replacing `foo/bar` with invalid name
```

Not only will this test more invalid names, but it will make it easier to test other commands that take volumes.